src/utils/currency/currency.ts

```diff
@@ -1,115 +1,57 @@
-
-const buildCoinGeckoURL = (
-  contractAddress: string,
-  chainId: SupportedChains = Network.Gnosis,
-  conversionDenomination: CoinGeckoSupportedCurrencies = SupportedCurrencies.Usd,
-) => {
-  const chain = mapToAPIFormat(chains, chainId);
-  const denomination = mapToAPIFormat(currencies, conversionDenomination);
-
-  return buildAPIEndpoint(new URL(`${currencyApiConfig.endpoint}${chain}`), {
-    [currencyApiConfig.searchParams.from]: contractAddress,
-    [currencyApiConfig.searchParams.to]: denomination,
-    [currencyApiConfig.searchParams.api]: process.env.COINGECKO_API_KEY ?? '',
-  });
-};
-
-const extractPriceFromCoinGeckoResponse = (
-  data: CoinGeckoPriceRequestSuccessResponse,
-  contractAddress: string,
-  conversionDenomination: CoinGeckoSupportedCurrencies,
-) => {
-  try {
-    return data[contractAddress.toLowerCase()][
-      mapToAPIFormat(currencies, conversionDenomination)
-    ];
-  } catch (e) {
-    console.error(
-      'Could not get price from CoinGecko response. Response shape might have changed.',
-      e,
-    );
-    return 0;
-  }
-};
-
-const isCoinGeckoSuccessResponse = (
-  data: CoinGeckoPriceRequestResponse,
-): data is CoinGeckoPriceRequestSuccessResponse => {
-  if (
-    typeof data === 'undefined' || // undefined
-    data == null || // null
-    (typeof data == 'object' && Object.keys(data).length === 0) // empty object
-  ) {
-    return false;
-  }
-
-  return true;
-};
-
 const fetchPriceFromCoinGecko = async ({
   contractAddress,
   chainId = Network.Gnosis,
-  conversionDenomination = SupportedCurrencies.Usd,
+  conversionDenomination,
 }: Pick<FetchCurrentPriceArgs, 'chainId' | 'contractAddress'> & {
-  conversionDenomination?: CoinGeckoSupportedCurrencies;
+  conversionDenomination: CoinGeckoSupportedCurrencies;
 }) => {
-  const url = buildCoinGeckoURL(
-    contractAddress,
-    chainId,
-    conversionDenomination,
-  );
-
-  const price = await fetchData<CoinGeckoPriceRequestResponse, number>(
-    url,
-    (data) => {
-      if (isCoinGeckoSuccessResponse(data)) {
-        return extractPriceFromCoinGeckoResponse(
-          data,
-          contractAddress,
-          conversionDenomination,
-        );
-      }
+  // If it's a network token we can't fetch it by contract address
+  if (contractAddress === ADDRESS_ZERO) {
+    const networkToken =
+      coinGeckoMappings.networkTokens[DEFAULT_NETWORK_TOKEN.symbol];
 
-      console.error(
-        `Unable to get price for ${contractAddress}. It probably doesn't have a listed exchange value.`,
-      );
+    if (!networkToken) {
+      console.error('Unable to get default network token.');
       return 0;
-    },
-    `Api called failed at ${url}.`,
-  );
+    }
 
-  return price;
+    return fetchTokenPrice({
+      tokenName: networkToken,
+      conversionDenomination,
+    });
+  }
+
+  return fetchContractPrice({
+    contractAddress,
+    chainId,
+    conversionDenomination,
+  });
```

Comment:
Gotta nit-pick on naming here, so they're both fetching token price, but one is doing so by name, other by address, right? Could the naming reflect that?